src/hotspot/share/opto/countbitsnode.cpp

```diff
@@ -26,25 +26,37 @@
 #include "opto/opcodes.hpp"
 #include "opto/phaseX.hpp"
 #include "opto/type.hpp"
+#include "utilities/count_leading_zeros.hpp"
+#include "utilities/count_trailing_zeros.hpp"
+
+static int count_leading_zeros_int(jint i) {
+  return i == 0 ? BitsPerInt : count_leading_zeros(i);
+}
+
+static int count_leading_zeros_long(jlong l) {
+  return l == 0 ? BitsPerLong : count_leading_zeros(l);
+}
+
+static int count_trailing_zeros_int(jint i) {
+  return i == 0 ? BitsPerInt : count_trailing_zeros(i);
+}
+
+static int count_trailing_zeros_long(jlong l) {
+  return l == 0 ? BitsPerLong : count_trailing_zeros(l);
+}
```

Comment:
Can you explain why you need this?
Why is `count_trailing_zeros` and `count_leading_zeros` not enough, when you cast at the use-site?

Reply:
This is because our implementation does not accept 0 as an input. I suggest doing this at `count_leading_zeros`, it makes more sense and also aligns our behaviour with the well-known `[countr_zero](https://en.cppreference.com/w/cpp/numeric/countr_zero.html)` and `[countl_zero](https://en.cppreference.com/w/cpp/numeric/countl_zero.html)`